src/tcutility/analysis/report/report.py

```diff
@@ -0,0 +1,111 @@
+import pathlib as pl
+from typing import Union
+
+import docx
+from htmldocx import HtmlToDocx
+from tcutility.analysis.report.formatters.generic import WordFormatter
+from tcutility.analysis.report.formatters.xyz import XYZFormatter
+from tcutility.results import Result, read
+
+
+class SI:
+    def __init__(self, path: Union[str, pl.Path], append_mode: bool = False, font: str = "Arial", format: WordFormatter = XYZFormatter()) -> None:
```

Comment:
the format argument here now defaults to the XYZformatter. The issue I see with this is that an SI can be more than only the xyz-coordinates. For example, we might want to add tables, figures, text, etc. It might be better for each type of section to have its own formatter. For example, the SI class can have the `xyz_formatter`, `table_formatter`, `figure_formatter`, etc. arguments. This way we can easily swap between different styles of tables, and xyz-coordinate sections.

Reply:
Aha I see what you mean and I agree about that we should take into account different ways of writing the SI such as a table or figure formatter. Do you have ideas about what the arguments are when trying to write a table or figure? 

I say this because one option is to introduce methods including "add_table", "add_figure", each having a default formatter similar to the xyz formatter.